utils/src/main/java/software/amazon/awssdk/utils/CompletableFutureUtils.java

```diff
@@ -76,4 +77,27 @@ public static CompletionException errorAsCompletionException(Throwable t) {
         });
         return src;
     }
+
+    /**
+     * Completes the {@code dst} future based on the result of the {@code src} future asynchronously on
+     * the provided {@link Executor} and return the {@code dst} future.
+     *
+     * @param src The source {@link CompletableFuture}
+     * @param dst The destination where the {@code Throwable} or response will be forwarded to.
+     * @param executor the executor to complete the des future
+     * @return {@code dst}.
+     */
+    public static <T> CompletableFuture<T> forwardResultTo(CompletableFuture<T> src,
+                                                           CompletableFuture<T> dst,
+                                                           Executor executor) {
+        src.whenCompleteAsync((r, e) -> {
+            if (e != null) {
+                dst.completeExceptionally(e);
+            } else {
+                dst.complete(r);
+            }
+        }, executor);
+
+        return dst;
```

Comment:
Returning `dst` seems intuitive to me, but `forwardExceptionTo` returns `src`. We are probably safer off being consistent.

Reply:
Yeah, sure